src/librustdoc/html/highlight.rs

```diff
@@ -155,6 +167,68 @@ impl<'a> Classifier<'a> {
             in_macro: false,
             in_macro_nonterminal: false,
             edition,
+            byte_pos: 0,
+            src,
+        }
+    }
+
+    /// Concatenate colons and idents as one when possible.
+    fn get_full_ident_path(&mut self) -> Vec<(TokenKind, usize, usize)> {
```

Comment:
How about ArrayVec? librustdoc already depends on it, and these Vecs have exactly zero, one, or two items in them.

Reply:
Good idea!

Reply:
Actually just gave it a try, no difference and it's actually not great in case we have less than 2 elements. Keeping it as is for the moment. If you want to give it another try later on, don't hesitate!